Debug builds

waych opened the pull request:
sdl2-sys was hardcoding the bundled build as a release build,
unconditionally for a long time.  This can cause problems however if an
upstream link of a binary is mixing C++ libraries that mix debug and non
debug builds, or so at least I'm led to believe for static builds.

Cobrand commented:
I think this PR caused  #1088, would you mind checking it out? If you don't have time to dig into this, I will have to revert this commit temporarily until we can find a way to include SDL2 debug builds that don't break anything else.

waych commented:
Taking a look see.

waych commented:
Let's rollback.  This needs more work.

It seems the use of the 'd' suffix may be tied to OS, setting it unconditionally at least doesn't seem to do the right thing in all cases.

Build works fine for me locally with both `["bundled"]` and `["bundled", "static-link"]`, but I added Windows bundled builds to the CI in the same commit which is producing link errors even after I fixed the broken file copy. I haven't yet been able to reproduce these errors locally.

Sorry for the break :(

waych commented:
Oh hmm. I was testing "bundled" with `["gfx", "image", "ttf", "mixer"]` set, but am realizing now that those aren't actually supported with "bundled" yet.
